src/main/java/com/dynamic/Fibonacci.java

```diff
@@ -0,0 +1,33 @@
+package src.main.java.com.dynamic;
+
+import java.util.HashMap;
+
+/**
+ * The Fibonacci sequence is a sequence of numbers in which each number after the first two is the sum of the two preceding ones.
+ * 1, 1, 2, 3, 5, 8, 13, ...
+ */
+public class Fibonacci {
+
+    private static HashMap<Integer, Integer> fibonacciCache = new HashMap<>();
```

Comment:
Instead of a HashMap, we can do this with constant space by using two variables.

Reply:
Also, An Array or ArrayList can be used instead of HashMap. Because the index of array can be assumed as the position of fibbonaci number :
index of Array- fibbonaci number
0 - 1
1 - 1
2 - 2
3 - 3
4 - 5
5 - 8
So, nth fibonacci number will be present at nth index of array.